### PR TITLE
Call idn_to_ascii with INTL_IDNA_VARIANT_UTS46

### LINKED ISSUE #1699: `idn_to_ascii(): INTL_IDNA_VARIANT_2003 is deprecated` in PHP 7.2
```PHP Deprecated:  idn_to_ascii(): INTL_IDNA_VARIANT_2003 is deprecated in FreshRSS/lib/lib_rss.php on line 65```

From https://secure.php.net/manual/en/function.idn-to-ascii.php I believe this change to https://github.com/FreshRSS/FreshRSS/blob/ceda55c75b158fc1cf4813fe0f258527754b9289/lib/lib_rss.php#L65 should work:
`$puny = idn_to_ascii($idn, 0, INTL_IDNA_VARIANT_UTS46);`

### PR DESCRIPTION
**Pull request by candrews**
Under PHP 7.2, calling `idn_to_ascii($idn)` results in a deprecation warning: 'INTL_IDNA_VARIANT_2003 is deprecated'
See https://secure.php.net/manual/en/function.idn-to-ascii.php 

Therefore, if possible, `idn_to_ascii($idn, 0, INTL_IDNA_VARIANT_UTS46)` should be used instead. `INTL_IDNA_VARIANT_UTS46` was introduced in PHP 5.4, so on versions before that, `idn_to_ascii($idn)` must still be used.

Fixed #1699

### CODE REVIEW

**Review by Frenzie — APPROVED** [review]
Lgtm. Btw, I wonder if maintaining 5.3 support is still worthwhile?

File: lib/lib_rss.php [hunk 1]
```diff
@@ -62,6 +62,12 @@ function idn_to_puny($url) {
 		$parts = parse_url($url);
 		if (!empty($parts['host'])) {
 			$idn = $parts['host'];
+			// INTL_IDNA_VARIANT_UTS46 is defined starting in PHP 5.4
+			if (defined('INTL_IDNA_VARIANT_UTS46')) {
+				$puny = idn_to_ascii($idn, 0, INTL_IDNA_VARIANT_UTS46);
+			} else {
+				$puny = idn_to_ascii($idn);
+			}
 			$puny = idn_to_ascii($idn);
```
Comment (Alkarex) [inline lib/lib_rss.php]: This line seems to have been forgotten

**Comment by Alkarex** [comment]
Please target the /dev branch. I have rebased from the GitHub interface, ~~but this included a couple of unrelated commits~~. edit: fixed :-)

**Comment by Frenzie** [comment]
> Please target the /dev branch.

Oh drat.

**Comment by Alkarex** [comment]
And please add a line for you in https://github.com/FreshRSS/FreshRSS/blob/dev/CREDITS.md (only if you want) :-)
Edit: Done :-)

**Pull request merged**